File: 2125.number-of-laser-beams-in-a-bank/solution.go
func numberOfBeams(bank []string) int {
    res := 0
    prev := 0
    for _, row := range bank {
        devices := 0
        for _, cell := range row {
            if cell == '1' {
                devices++
            }
        }
        if devices > 0 {
            res += prev * devices
            prev = devices
        }
    }
    return res
}

File: 2125.number-of-laser-beams-in-a-bank/solution.kt
class Solution {
    fun numberOfBeams(bank: Array<String>): Int {
        var res = 0
        var previous = 0
        bank.forEach { row ->
            val devices = row.count { it == '1' }
            if (devices > 0) {
                res += previous * devices
                previous = devices
            }
        }
        return res
    }
}